kettle/kettle.py

```diff
@@ -267,8 +267,11 @@ def handle(self):
 
 		while True:
```

Comment:
Instead, you could do `while manager.game.step != Step.FINAL_GAMEOVER`, and some extra logic at the end of that block.